tools/install.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+# Copyright 2020 The Emscripten Authors.  All rights reserved.
+# Emscripten is available under two separate licenses, the MIT license and the
+# University of Illinois/NCSA Open Source License.  Both these licenses can be
+# found in the LICENSE file.
+
+"""Install the parts of emscripten needed for end users. This works like
+a traditional `make dist` target but is writtin python to it can be portable
+and run on non-unix platforms (basically windows).
+"""
+
+from __future__ import print_function
+
+import argparse
+import fnmatch
+import os
+import shutil
+import subprocess
+import sys
+
+EXCLUDES = '''
+tests/third_party
+site
+node_modules
+docs
+Makefile
+.git
+'''.split()
+
+EXCLUDE_PATTERNS = '''
+*.pyc
+.*
+'''.split()
+
+EXCLUDE_DIRS = '''
+__pycache__
+'''.split()
+
+
+def add_revision_file(target):
+  # text=True would be better than encoding here, but its only supported in 3.7+
+  git_hash = subprocess.check_output(['git', 'rev-parse', 'HEAD'], encoding='utf-8').strip()
```

Comment:
why is an encoding needed, the output should just be ascii?

Reply:
With python3 subprocess output is always `bytes` unless you set with `text=True` (which we can't, see above) or set some encoding.